test/blackbox/common/BlackboxTestsTransportCustom.cpp

```diff
@@ -188,3 +188,121 @@ TEST(ChainingTransportTests, basic_test)
+            {
+                times_reader_receive_function_called.fetch_add(1);
+            };
+    reader_transport->send_function_called = [&times_reader_send_function_called]()
+            {
+                times_reader_send_function_called.fetch_add(1);
+            };
+
+    std::shared_ptr<eprosima::fastdds::rtps::TCPv4TransportDescriptor> writer_tcp_transport =
+            std::make_shared<eprosima::fastdds::rtps::TCPv4TransportDescriptor>();
+
+    std::shared_ptr<TestChainingTransportDescriptor> writer_transport =
+            std::make_shared<TestChainingTransportDescriptor>(writer_tcp_transport);
+    writer_transport->init_function_called = [&times_writer_init_function_called]()
+            {
+                times_writer_init_function_called.fetch_add(1);
+            };
+    writer_transport->receive_function_called = [&times_writer_receive_function_called]()
+            {
+                times_writer_receive_function_called.fetch_add(1);
+            };
+    writer_transport->send_function_called = [&times_writer_send_function_called]()
+            {
+                times_writer_send_function_called.fetch_add(1);
+            };
+
+    PubSubWriter<HelloWorldPubSubType> writer(TEST_TOPIC_NAME);
+    PubSubReader<HelloWorldPubSubType> reader(TEST_TOPIC_NAME);
+
+    eprosima::fastrtps::rtps::LocatorList_t initial_peers;
+    initial_peers.push_back(reader_loc);
+
+    writer.disable_builtin_transport()
+            .add_user_transport_to_pparams(writer_transport)
+            .initial_peers(initial_peers)
+            .history_depth(10)
+            .property_policy(test_property_policy)
+            .init();
+
+    ASSERT_TRUE(writer.isInitialized());
+
+    reader.disable_builtin_transport()
+            .add_user_transport_to_pparams(reader_transport)
+            .reliability(eprosima::fastrtps::RELIABLE_RELIABILITY_QOS)
+            .property_policy(test_property_policy)
+            .metatraffic_unicast_locator_list(reader_locators)
+            .set_default_unicast_locators(reader_locators)
+            .init();
+
+    ASSERT_TRUE(reader.isInitialized());
+
+    // Wait for discovery.
+    writer.wait_discovery();
+    reader.wait_discovery();
+
+    auto data = default_helloworld_data_generator(1);
+    reader.startReception(data);
+    writer.send(data);
+    ASSERT_TRUE(data.empty());
+    reader.block_for_all();
+
+    ASSERT_EQ(times_writer_init_function_called.load(), 1);
+    ASSERT_EQ(times_reader_init_function_called.load(), 1);
+    ASSERT_GE(times_writer_send_function_called.load(), 0);
+    ASSERT_GE(times_reader_receive_function_called.load(), 0);
+
+    //! If only 1 sender resource was created
+    //! Expect less than 30 calls in send/receive
+    //! including discovery phase calls and reception.
+    //! Else something is wrong, more than one sender resource
+    //! is being created
+    ASSERT_LE(times_writer_send_function_called.load(), 30);
```

Comment:
Why this number 30?

Reply:
If more sender resources than expected are created, the `send()` function call will be called in a hundred-magnitude of times (experimentally observed for this test). Otherwise, if the test successfully creates only one sender, then it will be called (~20 times). So, a good threshold taking into account that in some platforms the execution can be slower, and faster in others, can be 30. 
But yes, it is not "analytically" deduced, but the result of bearing in mind the test behavior, the platform and the addition of a little padding margin.